packages/host/app/resources/module-contents.ts

```diff
@@ -0,0 +1,180 @@
+import { tracked } from '@glimmer/tracking';
+
+import { Resource } from 'ember-resources';
+
+import { getAncestor, getField } from '@cardstack/runtime-common';
+
+import {
+  ModuleSyntax,
+  type PossibleCardOrFieldClass,
+  type BaseDeclaration,
+  type Declaration,
+  isInternalReference,
+  isPossibleCardOrFieldClass,
+} from '@cardstack/runtime-common/module-syntax';
+
+import {
+  getCardType,
+  type CardType,
+} from '@cardstack/host/resources/card-type';
+
+import { Ready as ReadyFile } from '@cardstack/host/resources/file';
+
+import { type BaseDef } from 'https://cardstack.com/base/card-api';
+
+// an declaration should be (an item of focus within a module)
+// - exported function or class
+// - exported card or field
+// - unexported card or field
+// This declaration (in code mode) is extended to include the cardType and cardOrField
+export type ModuleDeclaration =
+  | (CardOrField & Partial<PossibleCardOrFieldClass>)
+  | BaseDeclaration;
+
+export interface CardOrField {
+  cardType: CardType;
+  cardOrField: typeof BaseDef;
+}
+
+export function isCardOrFieldDeclaration(
+  declaration: ModuleDeclaration,
+): declaration is CardOrField & Partial<PossibleCardOrFieldClass> {
+  return (
+    (declaration as CardOrField).cardType !== undefined &&
+    (declaration as CardOrField).cardOrField !== undefined
+  );
+}
+
+interface Args {
+  named: { file: ReadyFile; exportedCardsOrFields: (typeof BaseDef)[] };
+}
+
+export class ModuleContentsResource extends Resource<Args> {
+  @tracked private _declarations: ModuleDeclaration[] = [];
+
+  get declarations() {
+    return this._declarations;
+  }
+
+  modify(_positional: never[], named: Args['named']) {
+    let { file, exportedCardsOrFields } = named;
+    let moduleSyntax = new ModuleSyntax(file.content);
+    let localCardsOrFields = collectLocalCardsOrFields(
+      moduleSyntax,
+      exportedCardsOrFields,
+    );
+
+    // This loop
+    // - adds card type (not loaded tho)
+    // - includes card or field
+    //   - an already exported card or field
+    //   - an already that was local but exported thru some relationship
```

Comment:
Looks like something is missing in the comment - an already what?

Reply:
https://github.com/cardstack/boxel/pull/716/commits/798bb26b54cce6777a296f49c3dd321ddfe6d761 

This was poor english